controllers/baremetalset_controller.go

```diff
@@ -440,13 +473,16 @@ func (r *BaremetalSetReconciler) baremetalHostProvision(instance *ospdirectorv1b
 	}
 
 	sts = append(sts, userDataSt)
+	ip, network, _ := net.ParseCIDR(ipset.Status.HostIPs[bmhName].IPAddresses["ctlplane"]) // We use ctlplane as the MgmtNetwork too for now
+
+	r.Log.Info(fmt.Sprintf("Network: %s", network.String()))
 
 	// Network data cloud-init secret
 	templateParameters = make(map[string]string)
-	templateParameters["MgmtIp"] = mgmtIP
+	templateParameters["MgmtIp"] = ip.String()
 	templateParameters["MgmtInterface"] = instance.Spec.MgmtInterface
-	templateParameters["MgmtGateway"] = gwIP
-	templateParameters["MgmtNetmask"] = netmask
+	templateParameters["MgmtGateway"] = "192.168.1.1" //FIXME
+	templateParameters["MgmtNetmask"] = fmt.Sprintf("/%s", strings.Split(network.String(), "/")[1])
```

Comment:
This format might not work properly with RHEL's cloud-init, unless you've tried it and found it to be okay.  I say that only because the examples I found all used `X.X.X.X` notation for IPv4 (never found an IPv6 example, however).

Reply:
let me research this again. I meant to do this as a separate patch actually and it slipped in on a rebase. Probably best to just leave it as a dotted quad format

Reply:
Latest patch has this resolved.